src/Components/Web/src/Forms/InputRadioGroup.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// Groups child <see cref="InputRadio{TValue}"/> components.
+    /// </summary>
+    public class InputRadioGroup : ComponentBase
+    {
+        internal string? GroupName { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the child content to be rendering inside the <see cref="InputRadioGroup"/>.
+        /// </summary>
+        [Parameter] public RenderFragment? ChildContent { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the group.
+        /// </summary>
+        [Parameter] public string? Name { get; set; }
+
+        /// <inheritdoc />
+        protected override void OnParametersSet()
+        {
+            GroupName ??= !string.IsNullOrEmpty(Name) ? Name : Guid.NewGuid().ToString("N");
```

Comment:
Setting any parameters on this component would generate a new guid, which would cause a larger delta change to the DOM when all inputs inside it need to update their names. 

Would it be better to assign a new guid in a private member, and use that here instead?

There aren't any additional parameters at the moment, but if someone descends from it to add functionality there will be.

Reply:
Thanks for the feedback! In this case, the `??=` operator prevents reassignment of `GroupName` and the regeneration of the GUID, but your suggestion does allow the user to update the `Name` if provided explicitly.

Reply:
Sorry, not yet used to spotting the new operators, I've worked for so many decades without them. Your code is just fine.